hera_cal/lstbin.py

```diff
@@ -0,0 +1,893 @@
+        on the real and imag components separately. Warning: This considerably slows down the code.
+
+    sigma : type=float, input standard deviation to use for sigma clipping algorithm.
+
+    return_no_avg : type=boolean, if True, return binned but un-averaged data and wgts.
+
+    Output: (data_avg, wgts_avg, data_std, lst_bins, data_num)
+    -------
+    data_avg : dictionary of data having averaged the LST bins
+
+    wgts_avg : dictionary of data weights averaged in LST bins
+    
+    data_std : dictionary of data with real component holding std along real axis
+        and imag component holding std along imag axis
+
+    lst_bins : ndarray containing final lst grid of data
+
+    data_num : dictionary containing the number of data points averaged in each LST bin.
+
+    if return_no_avg:
+        Output: (data_bin, wgts_bin)
+        data_bin : dictionary with (ant1,ant2,pol) as keys and ndarrays holding
+            un-averaged complex visibilities in each LST bin as values. 
+    """
+    # construct lst_grid if not provided
+    if lst_grid is None:
+        # get dlst if not provided
+        if dlst is None:
+            dlst = np.median(np.diff(lst_list[0]))
+        # construct lst_grid
+        lst_grid = np.arange(lst_init, lst_init + wrap_point - dlst/2, dlst) + dlst/2
+    else:
+        dlst = np.median(np.diff(lst_grid))
+        lst_init = lst_grid[0] - dlst/2
+        if lst_init < 0:
+            lst_init += wrap_point
+
+    # restrict lst_grid based on lst_low and lst_high
+    if lst_low is not None:
+        lst_grid = lst_grid[lst_grid > lst_low - atol]
+    if lst_hi is not None:
+        lst_grid = lst_grid[lst_grid < lst_hi + atol]
+
+    # move lst_grid centers to the left
+    lst_grid_left = lst_grid - dlst / 2
+    lst_grid_left = unwrap(lst_grid_left)
+
+    # form new dictionaries
+    # data is a dictionary that will hold other dictionaries as values, which will
+    # themselves hold lists of ndarrays
+    data = odict()
+    wgts = odict()
+    all_lst_indices = set()
+
+    # iterate over data_list
+    for i, d in enumerate(data_list):
+        # get lst array
+        l = lst_list[i]
+
+        # unwrap LST array
+        l = unwrap(l)
+
+        # add wrap_point to l where less than lst_init
+        l[np.where(l < lst_init)] += wrap_point
+
+        # digitize data lst array "l", moving bin centers to the left
+        grid_indices = np.digitize(l, lst_grid_left[1:], right=True)
+
+        # make data_in_bin boolean array, and set to False data that don't fall in any bin
+        data_in_bin = np.ones_like(l, np.bool)
+        data_in_bin[(l<lst_grid_left.min()-atol)|(l>lst_grid_left.max()+dlst+atol)] = False
+
+        # update all_lst_indices
+        all_lst_indices.update(set(grid_indices[data_in_bin]))
+
+        # iterate over keys in d
+        for j, key in enumerate(d.keys()):
+            # conjugate d[key] if necessary
+            if key in data:
```

Comment:
All the conjugation stuff seems unnecessary since we are using DataContainers as the input data type.

Reply:
can't use DataContainer here, so this is necessary

Reply:
Bump...we haven't figured this out yet.